tests_integration/spcs/test_registry.py

```diff
@@ -11,3 +14,33 @@ def test_token(runner):
     assert result.json["token"]
     assert "expires_in" in result.json
     assert result.json["expires_in"]
+
+
+@pytest.mark.integration
+def test_get_registry_url(test_database, test_role, runner, snowflake_session):
+    # newly created test_role should have no access to image repositories and should not be able to get registry URL
+    test_repo = ObjectNameProvider("test_repo").create_and_get_next_object_name()
+    snowflake_session.execute_string(f"create image repository {test_repo}")
+
+    fail_result = runner.invoke_with_connection(
+        ["spcs", "image-registry", "url", "--role", test_role]
+    )
+    assert fail_result.exit_code == 1, fail_result.output
+    assert "No image repository found." in fail_result.output
+
+    # role should be able to get registry URL once granted read access to an image repository
+    repo_list_cursor = snowflake_session.execute_string(
+        "show image repositories", cursor_class=DictCursor
+    )
+    expected_repo_url = repo_list_cursor[0].fetchone()["repository_url"]
```

Comment:
In `test_utils.py` are functions to extract data from snowflake session result. In this case you could use `row_from_snowflake_session`

Reply:
done!